src/aiofiles/threadpool/__init__.py

```diff
@@ -83,26 +100,71 @@ def _open(
 
 
 @singledispatch
-def wrap(file, *, loop=None, executor=None):
+def wrap(file, *, loop=None, executor=None, indirect=None):
     raise TypeError("Unsupported io type: {}.".format(file))
 
 
 @wrap.register(TextIOBase)
-def _(file, *, loop=None, executor=None):
-    return AsyncTextIOWrapper(file, loop=loop, executor=executor)
+def _(file, *, loop=None, executor=None, indirect=None):
+    if indirect is None:
+        return AsyncTextIOWrapper(file, loop=loop, executor=executor)
+    else:
+        return AsyncTextIndirectIOWrapper(
+            file, loop=loop, executor=executor, indirect=indirect
+        )
 
 
 @wrap.register(BufferedWriter)
-def _(file, *, loop=None, executor=None):
-    return AsyncBufferedIOBase(file, loop=loop, executor=executor)
+@wrap.register(BufferedIOBase)
+def _(file, *, loop=None, executor=None, indirect=None):
+    if indirect is None:
+        return AsyncBufferedIOBase(file, loop=loop, executor=executor)
+    else:
+        return AsyncIndirectBufferedIOBase(
+            file, loop=loop, executor=executor, indirect=indirect
+        )
 
 
 @wrap.register(BufferedReader)
 @wrap.register(BufferedRandom)
-def _(file, *, loop=None, executor=None):
-    return AsyncBufferedReader(file, loop=loop, executor=executor)
+def _(file, *, loop=None, executor=None, indirect=None):
+    if indirect is None:
+        return AsyncBufferedReader(file, loop=loop, executor=executor)
+    else:
+        return AsyncIndirectBufferedReader(
+            file, loop=loop, executor=executor, indirect=indirect
+        )
 
 
 @wrap.register(FileIO)
-def _(file, *, loop=None, executor=None):
-    return AsyncFileIO(file, loop, executor)
+def _(file, *, loop=None, executor=None, indirect=None):
+    if indirect is None:
+        return AsyncFileIO(file, loop, executor)
+    else:
+        return AsyncIndirectFileIO(file, loop, executor, indirect=indirect)
+
+
+try:
+    stdin = wrap(sys.stdin, indirect=lambda: sys.stdin)
+except TypeError:
```

Comment:
Where do the TypeErrors come from btw?

Reply:
`sys.stdin` may have been patched to be an unusable base type prior to importing aiofiles. This happens specifically when using pytest. I figured I would add it to the rest of the stdio files just be safe.

Reply:
I see, alright. Do we want to document this for people somewhere?

Reply:
imo the best documentation would be replacing the `stdin = None` with a custom class which raises a descriptive error when you try to use any of its methods. I was hoping I could convince you to do that :sweat_smile: 

lmk if you would like to do that or just add a note to the readme/docs

Reply:
Never mind, just thought of a much better solution. One sec.